File: examples/xatBlogAPI-lib.go
package main

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
	"os"
	"strings"
)

type XatBlogAPI struct {
	BaseURL string
}

func XatBlogAPIData() *XatBlogAPI {
	return &XatBlogAPI{
		BaseURL: "https://api.xatblog.net/",
	}
}

func (x *XatBlogAPI) sendRequest(path string) (map[string]interface{}, error) {
	url := x.BaseURL + path

	// Desativa a verificação SSL
	tr := &http.Transport{
		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
	}
	client := &http.Client{Transport: tr}

	response, err := client.Get(url)
	if err != nil {
		return nil, fmt.Errorf("failed to send request: %s", err.Error())
	}
	defer response.Body.Close()

	body, err := ioutil.ReadAll(response.Body)
	if err != nil {
		return nil, fmt.Errorf("failed to read response body: %s", err.Error())
	}

	var data map[string]interface{}
	err = json.Unmarshal(body, &data)
	if err != nil {
		return nil, fmt.Errorf("failed to parse response JSON: %s", err.Error())
	}

	return data, nil
}

func (x *XatBlogAPI) latest() (map[string]interface{}, error) {
	return x.sendRequest("/latest")
}

func (x *XatBlogAPI) prices() (map[string]interface{}, error) {
	return x.sendRequest("/prices")
}

func (x *XatBlogAPI) countdown() (map[string]interface{}, error) {
	return x.sendRequest("/countdown")
}

func (x *XatBlogAPI) promoted() (map[string]interface{}, error) {
	return x.sendRequest("/promoted")
}

func (x *XatBlogAPI) activePawns() (map[string]interface{}, error) {
	return x.sendRequest("/activepawns")
}

func (x *XatBlogAPI) regToID(username string) (map[string]interface{}, error) {
	endpoint := "/reg2id/" + username
	return x.sendRequest(endpoint)
}

func (x *XatBlogAPI) idToReg(id string) (map[string]interface{}, error) {
	endpoint := "/id2reg/" + id
	return x.sendRequest(endpoint)
}

func (x *XatBlogAPI) namePrice(username string) (map[string]interface{}, error) {
	endpoint := "/nameprice/" + username
	return x.sendRequest(endpoint)
}

func (x *XatBlogAPI) chatPrice(chatname string) (map[string]interface{}, error) {
	endpoint := "/chatprice/" + chatname
	return x.sendRequest(endpoint)
}

func (x *XatBlogAPI) chatInfo(chatname string) (map[string]interface{}, error) {
	endpoint := "/chatinfo/" + chatname
	return x.sendRequest(endpoint)
}

func (x *XatBlogAPI) powerSearch(powername string) (map[string]interface{}, error) {
	endpoint := "/powersearch/" + powername
	return x.sendRequest(endpoint)
}

func (x *XatBlogAPI) powerInfo(powername string) (map[string]interface{}, error) {
	endpoint := "/powerinfo/" + powername
	return x.sendRequest(endpoint)
}

func (x *XatBlogAPI) powerLogs(powername string) (map[string]interface{}, error) {
	endpoint := "/powerlogs/" + powername
	return x.sendRequest(endpoint)
}

func (x *XatBlogAPI) promoPrice(hours int, language string) (map[string]interface{}, error) {
	endpoint := fmt.Sprintf("/promoprice/%d/%s", hours, language)
	return x.sendRequest(endpoint)
}

func (x *XatBlogAPI) daysToXats(amount int) (map[string]interface{}, error) {
	endpoint := "/dx/" + strconv.Itoa(amount)
	return x.sendRequest(endpoint)
}

func (x *XatBlogAPI) xatsToDays(amount int) (map[string]interface{}, error) {
	endpoint := "/x2d/" + strconv.Itoa(amount)
	return x.sendRequest(endpoint)
}

func (x *XatBlogAPI) verifyBanner(url string) (map[string]interface{}, error) {
	endpoint := "/verifybanner/" + url
	return x.sendRequest(endpoint)
}

func (x *XatBlogAPI) userGifts(userOrID string) (map[string]interface{}, error) {
	endpoint := "/usergifts/" + userOrID
	return x.sendRequest(endpoint)
}

func (x *XatBlogAPI) jinxList(powername string) (map[string]interface{}, error) {
	var endpoint string
	if powername != "" {
		endpoint = "/jinxlist/" + powername
	} else {
		endpoint = "/jinxlist"
	}
	return x.sendRequest(endpoint)
}

func main() {
	xatAPI := XatBlogAPIData()

	latestData, err := xatAPI.latest()
	if err != nil {
		fmt.Println("Failed to retrieve latest data:", err)
		os.Exit(1)
	}
	fmt.Println("Latest data:", latestData)

	pricesData, err := xatAPI.prices()
	if err != nil {
		fmt.Println("Failed to retrieve prices data:", err)
		os.Exit(1)
	}
	fmt.Println("Prices data:", pricesData)
}

File: examples/xatBlogAPI-lib.cpp
#include <iostream>
#include <curl/curl.h>
#include <nlohmann/json.hpp>

using json = nlohmann::json;

class XatBlogAPI {
private:
    static constexpr const char* BASE_URL = "https://api.xatblog.net/";

    static size_t writeCallback(void* contents, size_t size, size_t nmemb, std::string* output) {
        size_t totalSize = size * nmemb;
        output->append(static_cast<char*>(contents), totalSize);
        return totalSize;
    }

    json sendRequest(const std::string& path) {
        std::string url = BASE_URL + path;
        std::string response;

        CURL* curl = curl_easy_init();
        if (curl) {
            curl_easy_setopt(curl, CURLOPT_URL, url.c_str());
            curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, writeCallback);
            curl_easy_setopt(curl, CURLOPT_WRITEDATA, &response);
            CURLcode res = curl_easy_perform(curl);
            if (res != CURLE_OK) {
                std::cerr << "Request failed: " << curl_easy_strerror(res) << std::endl;
            }
            curl_easy_cleanup(curl);
        }

        return json::parse(response);
    }

public:
    json latest() {
        return sendRequest("/latest");
    }

    json prices() {
        return sendRequest("/prices");
    }

    json countdown() {
        return sendRequest("/countdown");
    }

    json promoted() {
        return sendRequest("/promoted");
    }

    json activePawns() {
        return sendRequest("/activepawns");
    }

    json regToId(const std::string& username) {
        std::string endpoint = "/reg2id/" + username;
        return sendRequest(endpoint);
    }

    json idToReg(const std::string& id) {
        std::string endpoint = "/id2reg/" + id;
        return sendRequest(endpoint);
    }

    json namePrice(const std::string& username) {
        std::string endpoint = "/nameprice/" + username;
        return sendRequest(endpoint);
    }

    json chatPrice(const std::string& chatname) {
        std::string endpoint = "/chatprice/" + chatname;
        return sendRequest(endpoint);
    }

    json chatInfo(const std::string& chatname) {
        std::string endpoint = "/chatinfo/" + chatname;
        return sendRequest(endpoint);
    }

    json powerSearch(const std::string& powername) {
        std::string endpoint = "/powersearch/" + powername;
        return sendRequest(endpoint);
    }

    json powerInfo(const std::string& powername) {
        std::string endpoint = "/powerinfo/" + powername;
        return sendRequest(endpoint);
    }

    json powerLogs(const std::string& powername) {
        std::string endpoint = "/powerlogs/" + powername;
        return sendRequest(endpoint);
    }

    json promoPrice(int hours, const std::string& language = "en") {
        std::string endpoint = "/promoprice/" + std::to_string(hours) + "/" + language;
        return sendRequest(endpoint);
    }

    json daysToXats(int amount) {
        std::string endpoint = "/dx/" + std::to_string(amount);
        return sendRequest(endpoint);
    }

    json xatsToDays(int amount) {
        std::string endpoint = "/x2d/" + std::to_string(amount);
        return sendRequest(endpoint);
    }

    json verifyBanner(const std::string& url) {
        std::string endpoint = "/verifybanner/" + url;
        return sendRequest(endpoint);
    }

    json userGifts(const std::string& user_or_id) {
        std::string endpoint = "/usergifts/" + user_or_id;
        return sendRequest(endpoint);
    }

    json jinxList(const std::string& powername = "") {
        std::string endpoint = (powername.empty()) ? "/jinxlist" : "/jinxlist/" + powername;
        return sendRequest(endpoint);
    }
};
